tests/transaction.spec.ts

```diff
@@ -77,5 +149,85 @@ describe('transaction', () => {
 
     expect(tracker.history.insert).toHaveLength(1);
     expect(tracker.history.delete).toHaveLength(1);
+
+    const transactions = tracker.getTransactions();
+
+    expect(transactions).toEqual([{
+      id: 0,
+      state: 'commited',
+      queries: [
+        expect.objectContaining({
+          method: 'insert',
+        }),
+        expect.objectContaining({
+          method: 'delete',
+        }),
+      ],
+    }]);
+  });
+
+  it('should keep track of interleaving transactions', async () => {
+    tracker.on.any(/.*/).response(1);
+
+    const trx1 = await db.transaction();
+    const trx2 = await db.transaction();
+
+    await trx1('table_one').insert({ name: faker.name.firstName() });
+    await trx2('table_two').insert({ name: faker.name.firstName() });
+
+    const nestedTrx1 = await trx1.transaction();
+    const nestedTrx2 = await trx2.transaction();
+
+    await nestedTrx1('table_one').delete().where({ name: faker.name.firstName() });
+    await nestedTrx2('table_two').delete().where({ name: faker.name.firstName() });
+
+    await nestedTrx2.commit();
+    await nestedTrx1.commit();
+
+    await trx1.commit();
+    await trx2.commit();
+
+    const transactions = tracker.getTransactions();
+
+    expect(transactions).toEqual([
+      {
+        id: 0,
+        state: 'commited',
+        queries: [
+          expect.objectContaining({
+            sql: 'insert into "table_one" ("name") values (?)',
+          }),
+        ],
+      },
+      {
+        id: 1,
+        state: 'commited',
+        queries: [
+          expect.objectContaining({
+            sql: 'insert into "table_two" ("name") values (?)',
+          }),
+        ],
+      },
+      {
+        id: 2,
+        parent: 0,
+        state: 'commited',
+        queries: [
+          expect.objectContaining({
+            sql: 'delete from "table_one" where "name" = ?',
```

Comment:
Is this should be `delete from table_two`? since `nestedTrx2` is committed first...

Reply:
I added a comment to indicate each transaction and mixed rollbacks with commits to include it on the demonstration.

Reply:
I'm not a big user of transactions, therefore, it was weird to me, if this is the correct behavior let's close it.